src/lightning/pytorch/callbacks/model_checkpoint.py

```diff
@@ -514,19 +514,23 @@ def _format_checkpoint_name(
 
         # check and parse user passed keys in the string
         groups = re.findall(r"(\{.*?)[:\}]", filename)
-        if len(groups) >= 0:
```

Comment:
isn't this needed anymore and why?

Reply:
Since groups is always a list, `if len(groups) >= 0` is equivalent to `if True`.

I have thought about changing to `if len(groups) > 0` instead. However, both the for loop and `str.format` works well when group is 0. If there is no curly bracket in the filename, the `format` method has no effect.

```python
filename = "best_model"  # groups = [] in this case
assert filename.format({})  == "best_model"
assert filename.format({"a": 1}) == "best_model"
```